code/code/misc/offense.cc

```diff
@@ -766,332 +766,301 @@ static bool canFleeThisWay(TBeing *ch, dirTypeT dir)
+  */
+  dirTypeT dirToUse = panic || chosenDir == DIR_NONE ||
+                              (enemy && ((doesKnowRetreat && !wasRetreatSuccessful) ||
+                                         (!doesKnowRetreat && ::number(0, 1))))
+                          ? randomDir
+                          : chosenDir;
+
+  // These messages don't make sense if not fighting
+  if (enemy) {
+    if (wasRetreatSuccessful && !panic) {
+      act("$n is looking for an opening to escape!", true, this, nullptr, nullptr, TO_ROOM);
+      act("You look for an opening to escape from the fight.", true, this, nullptr, nullptr,
+          TO_CHAR);
+    } else {
+      act("$n panics and attempts to flee!", true, this, nullptr, nullptr, TO_ROOM);
+      act("You turn tail and attempt to run away.", true, this, nullptr, nullptr, TO_CHAR);
+      loseSneak();
 
-  if (!(vict = fight())) {
-    for (i = 0; i < 20; i++) {
-      dirTypeT attempt = dirTypeT(::number(MIN_DIR, MAX_DIR-1));        // Select a random direction 
-      
-      // not fighting, so encourage flight to be in dir PC selected
-      if (chosenDir != DIR_NONE && !panic) {
-	attempt = chosenDir;
-      }
-      if (canFleeThisWay(this, attempt)) {
-        act("$n panics, and attempts to flee.", TRUE, this, 0, 0, TO_ROOM);
-        TBeing *tbt = dynamic_cast<TBeing *>(rider);
-        if (tbt) {
-          act("You turn tail and attempt to run away.", 
-	      TRUE, this, 0, 0, TO_CHAR);
-          loseSneak();
-          iDie = tbt->moveOne(attempt);
-          if (IS_SET_DELETE(iDie, DELETE_THIS)) {
-            tbt->reformGroup();
-            delete tbt;
-            tbt = NULL;
-            return FALSE;
-          }
-          if (iDie == TRUE) {
-            sendTo("You fled head over heels.\n\r");
-            return TRUE;
-          } else {
-            fleeFail(this);
-            return TRUE;
-          }
-        } else {
-          act("You turn tail and attempt to run away.", 
-	      TRUE, this, 0, 0, TO_CHAR);
-          loseSneak();
-          iDie = moveOne(attempt);
-          if (IS_SET_DELETE(iDie, DELETE_THIS))
-            return DELETE_THIS;
-	  
-          if (iDie == TRUE) {
-            sendTo(format("You nearly hurt yourself as you fled madly %swards.\n\r") % dirs[attempt]);
-            REMOVE_BIT(specials.affectedBy, AFF_ENGAGER);
-	    
-            return TRUE;
-          } else {
-            fleeFail(this);
-            return TRUE;
-          }
-        }
+      // Handle troglodyte racial
+      if (panic && !::number(0, 1) && getMyRace()->hasTalent(TALENT_MUSK) && getCond(FULL) > 5) {
+        act("Your fear causes you to release some musk scent to cover your tracks.", false, this, nullptr,
+            nullptr, TO_CHAR);
+        act("$n releases some musk into the room!", false, this, nullptr, nullptr, TO_ROOM);
+        dropGas(::number(1, 3), GAS_MUSK);
+        setCond(FULL, getCond(FULL) - 5);
```

Comment:
could have a consumeFullness or something. Not sure it'd be worth the extra lines tho.

Reply:
In interest of getting this PR done with I'm going to just agree with you and say it's not worth the extra lines. 😁